printer_zpl2/tests/test_printing_label_zpl2.py

```diff
@@ -0,0 +1,940 @@
+# Copyright 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import mock
+
+from odoo import exceptions
+from odoo.tests.common import TransactionCase
+
+
+model = 'odoo.addons.base_report_to_printer.models.printing_server'
+
+
+class TestPrintingLabelZpl2(TransactionCase):
+    def setUp(self):
+        super(TestPrintingLabelZpl2, self).setUp()
+        self.Model = self.env['printing.label.zpl2']
+        self.ComponentModel = self.env['printing.label.zpl2.component']
+        self.server = self.env['printing.server'].create({})
+        self.printer = self.env['printing.printer'].create({
+            'name': 'Printer',
+            'server_id': self.server.id,
+            'system_name': 'Sys Name',
+            'default': True,
+            'status': 'unknown',
+            'status_message': 'Msg',
+            'model': 'res.users',
+            'location': 'Location',
+            'uri': 'URI',
+        })
+        self.label_vals = {
+            'name': 'ZPL II Label',
+            'model_id': self.env.ref(
+                'base_report_to_printer.model_printing_printer').id,
+        }
+        self.component_vals = {
+            'name': 'ZPL II Label Component',
+        }
+
+    def new_label(self, vals=None):
+        values = self.label_vals.copy()
+        if vals is not None:
+            values.update(vals)
+        return self.Model.create(values)
+
+    def new_component(self, vals=None):
+        values = self.component_vals.copy()
+        if vals is not None:
+            values.update(vals)
+        return self.ComponentModel.create(values)
+
+    def test_print_on_bad_model(self):
+        """ Check that printing on the bad model raises an exception """
+        label = self.new_label()
+        with self.assertRaises(exceptions.UserError):
+            label.print_label(self.printer, label)
+
+    @mock.patch('%s.cups' % model)
+    def test_print_empty_label(self, cups):
+        """ Check that printing an empty label works """
+        label = self.new_label()
+        label.print_label(self.printer, self.printer)
+        cups.Connection().printFile.assert_called_once()
+
+    def test_empty_label_contents(self):
+        """ Check contents of an empty label """
+        label = self.new_label()
+        contents = label._generate_zpl2_data(self.printer)
+        self.assertEqual(
+            contents,
```

Comment:
this is a byte object, and so the comparison with the string is failing. 

TypeError: a bytes-like object is required, not 'str'

Reply:
Corresponded. Thank you.